src/Actions/PurchaseInvoice/CreateOrderFromPurchaseInvoice.php

```diff
@@ -65,4 +85,27 @@ public function prepareForValidation(): void
         $this->data = array_merge($this->purchaseInvoice?->toArray() ?? [], $this->data);
         $this->data['invoice_date'] ??= now()->toDateString();
     }
+
+    protected function validateData(): void
+    {
+        parent::validateData();
+
+        $errors = [];
+        if (
+            ! data_get($this->data, 'iban')
+            && app(PaymentType::class)->query()
+                ->whereKey(data_get($this->data, 'payment_type_id'))
+                ->value('requires_manual_transfer')
+            && ! app(ContactBankConnection::class)->query()
```

Comment:
No need for reversing the result. can just replace whereNotNull() with whereNull()